megamek/src/megamek/client/ui/swing/unitDisplay/WeaponPanel.java

```diff
@@ -1257,16 +1255,20 @@ public void displayMek(Entity en) {
 
         // change what is visible based on type
         if (entity.usesWeaponBays()) {
-            wArcHeatL.setVisible(true);
-            wArcHeatR.setVisible(true);
             m_chBayWeapon.setVisible(true);
             wBayWeapon.setVisible(true);
         } else {
-            wArcHeatL.setVisible(false);
-            wArcHeatR.setVisible(false);
             m_chBayWeapon.setVisible(false);
             wBayWeapon.setVisible(false);
         }
+        if (entity.isLargeCraft() && (game == null || !(game.getOptions().booleanOption(OptionsConstants.ADVAERORULES_HEAT_BY_BAY)))) {
```

Comment:
Is this correct?  The ArcHeat label will display if the game is null?

Reply:
Yeah, I think so. I originally didn't have that, but the code scan reported that the game may be null - This is at the beginning of the method:

```

// Grab a copy of the game.
Game game = null;

if (unitDisplay.getClientGUI() != null) {
  game = unitDisplay.getClientGUI().getClient().getGame();
}

```

Elsewhere in this method, when it wants to check the gameOptions, it first makes sure the game isn't null. If it is null, then it uses the default value:


```
 // Check Game Options for max external heat
int max_ext_heat = game != null ? game.getOptions().intOption(OptionsConstants.ADVCOMBAT_MAX_EXTERNAL_HEAT)
                : 15;

```

In this case, if the entity is a large craft, if we're unable to get the gameOption for ADVAERORULES_HEAT_BY_BAY we should assume the default setting - which means displaying heat by arc labels for large craft. I can change the block of code to be like this so it's using game !=null, but functionally it should be the exact same:


```
if ((!entity.isLargeCraft()) 
  || ((game != null) && (game.getOptions().booleanOption(OptionsConstants.ADVAERORULES_HEAT_BY_BAY)))){
  wArcHeatL.setVisible(false);
  wArcHeatR.setVisible(false);
}
else {
  wArcHeatL.setVisible(true);
  wArcHeatR.setVisible(true);
}

```

Thoughts?

Reply:
I see what you're saying, yeah.
I _personally_ prefer the latter implementation, `if ((!X) || ((Y != null) && (Y.z))`, from a readability standpoint but since it's functionally identical there's no reason to change it.

Reply:
To be fair, when I saw it & read your comment, I was questioning what I did too. I agree that it's more readable, and made the change. Thanks!